src/database/DataTypes/StocktakeItem.js

```diff
@@ -305,7 +305,16 @@ export class StocktakeItem extends Realm.Object {
    */
   get hasReason() {
     if (!this.batches) return false;
-    return this.batches.some(batch => batch.option);
+    return this.batches.every(batch => {
```

Comment:
While I say in `get hasAnyReason` I like just using `batch`, I think here destructuring into `{ snapshotQuanity, countedTotalQuantity, option }` could tidy this up:

```javascript
    return this.batches.every(({ snapshotTotalQuantity, countedTotalQuantity, option }) =>
      snapshotTotalQuantity !== countedTotalQuantity ? !!option : true
    );
```